### PR TITLE
#92 add potential fix for uniqueItems

### LINKED ISSUE #92: Generates incorrect code for array parameter with uniqueItems set to true
We have a parameter defined as follows where it takes a set of UUIDs:
 
        {
          "name" : "timeOffRequestIds",
          "in" : "body",
          "required" : true,
          "schema" : {
            "type" : "array",
            "items" : {
              "type" : "string",
              "format" : "uuid"
            },
            "uniqueItems" : true
          }
        }
        
 The parameter to the API is declared as a Set due to the uniqueItems:
 
     timeOffRequestIds: kotlin.collections.Set<kotlin.String>

But the code to serialize is trying to use `ListSerializer`:

     _json.encodeToString(ListSerializer(kotlin.String.serializer()), timeOffRequestIds)

That should be using `SetSerializer`, not `ListSerializer`

### PR DESCRIPTION
**Pull request by Tetraquark**
Closes #92

### CODE REVIEW

**Review by Alex009 — APPROVED** [review]

**Review by Alex009 — APPROVED** [review]

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt [hunk 1]
```diff
@@ -34,6 +34,80 @@ class ProfileApiTest {
...
+        val httpClient = createMockClient {
+            respondOk(content = testResponse)
+        }
+        val defaultApi = DefaultApi(httpClient = httpClient, json = json)
+
+        val result = runBlocking {
+            defaultApi.apiProfileGetMap()
+        }
+
+        assertTrue { result["user1"]!!.name == "user1" }
```
Comment (Alex009) [inline sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt:59]: here better will be:
```
assertEquals(
    expected = mapOf("user1" to User(name = "user1"), "user2" to User(name = "user2")),
    actual = result
)
```
but for now - ok

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt [hunk 2]
```diff
@@ -34,6 +34,80 @@ class ProfileApiTest {
...
+        }
+        val defaultApi = DefaultApi(httpClient = httpClient, json = json)
+
+        val result = runBlocking {
+            defaultApi.apiProfileGetSet()
+        }
+
+        assertNotNull(result.values)
+        assertTrue { result.values!!.size == 2 }
+        assertTrue { result.values!!.contains("str1") && result.values!!.contains("str2") }
```
Comment (Alex009) [inline sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt:82]: here better will be:
```
assertEquals(
    expected = listOf("str1", "str2"),
    actual = result.values
)
```
but for now - ok

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt [hunk 3]
```diff
@@ -34,6 +34,80 @@ class ProfileApiTest {
...
+                unitId = 1,
+                unitActivitySetItem = setOf(
+                    UnitActivitySetItem(activityRid = "1"),
+                    UnitActivitySetItem(activityRid = "2")
+                )
+            )
+        }
+
+        assertTrue { result.isNotEmpty() }
+        assertTrue { result.first().id == "10" }
```
Comment (Alex009) [inline sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt:110]: here better will be:
```
assertEquals(
    expected = setOf(Item(id = "10")),
    actual = result
)
```
but for now - ok

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt (hunk 1 above)
Comment (Tetraquark) [reply sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt]: Fixed

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt (hunk 2 above)
Comment (Tetraquark) [reply sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt]: Fixed

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt (hunk 3 above)
Comment (Tetraquark) [reply sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt]: Fixed

**Review by Alex009 — APPROVED** [review]

**Pull request merged**